Gems/PhysX/Code/Source/Common/PhysXSceneQueryHelpers.h

```diff
@@ -11,6 +11,7 @@
 #include<AzFramework/Physics/Common/PhysicsSceneQueries.h>
 
 //physx sdk includes
+#include <PxPhysicsVersion.h>
```

Comment:
This include, which I guess defines `PX_PHYSICS_VERSION_MAJOR` is only included here in `PhysXSceneQueryHelpers.h`, but the macro is used in many different cpps. Is it gurantee that all the cpps using the macro reaches `PhysXSceneQueryHelpers.h`? otherwise for some cpps `#if (PX_PHYSICS_VERSION_MAJOR == 5)` might be false and others true.

Maybe `#include <PxPhysicsVersion.h>` could be put in some basic header inside PhysX Gem to guarantee that?

Reply:
Good idea to try a non-unity build, all working fine with no issues.

There's no single basic header that is included everywhere (e.g. a precompiled header) but any code actually using PhysX SDK will have `PxPhysicsAPI.h` included which does include `PxPhysicsVersion.h` internally.